ui/server/src/main/scala/pl/touk/nussknacker/ui/api/NodesResources.scala

```diff
@@ -0,0 +1,48 @@
+package pl.touk.nussknacker.ui.api
+
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.Route
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import pl.touk.nussknacker.engine.ProcessingTypeData
+import pl.touk.nussknacker.engine.additionalInfo.{NodeAdditionalInfo, NodeAdditionalInfoProvider}
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.graph.node.NodeData
+import pl.touk.nussknacker.engine.util.loader.ScalaServiceLoader
+import pl.touk.nussknacker.ui.process.repository.FetchingProcessRepository
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.ui.process.ProcessingTypeDataProvider
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class NodesResources(val processRepository: FetchingProcessRepository[Future], typeToConfig: ProcessingTypeDataProvider[ProcessingTypeData])(implicit val ec: ExecutionContext)
+  extends ProcessDirectives with FailFastCirceSupport {
+
+  //TODO: do not load provider for each request...
+  private val providers: ProcessingTypeDataProvider[Option[NodeData => Future[Option[NodeAdditionalInfo]]]] = typeToConfig.mapValues(pt => ScalaServiceLoader
+    .load[NodeAdditionalInfoProvider](pt.modelData.modelClassLoader.classLoader).headOption.map(_.additionalInfo(pt.modelData.processConfig)))
+
+  def additionalData(implicit loggedUser: LoggedUser): Route = {
+    import akka.http.scaladsl.server.Directives._
+    import pl.touk.nussknacker.engine.graph.NodeDataCodec._
+
+    path("nodes" / Segment ) { processName =>
+      (post & processId(processName)) { processId =>
+        entity(as[NodeData]) { nodeData =>
+          complete {
+            processRepository.fetchLatestProcessDetailsForProcessId[CanonicalProcess](processId.id).map[ToResponseMarshallable] {
```

Comment:
Can we move this Business Logic to another component, like Service - I know, know ;) But still.. Let's move BL from adapters..

Reply:
Ok, I separated sth